### PR TITLE
Add CI step for linting

### PR DESCRIPTION
**Pull request by rbehjati**

### CODE REVIEW

**Comment by aferr** [comment]
You might need to rebase onto: #45

**Comment by aferr** [comment]
Ah, it looks like you may have already changed those, so I can drop #45 if you'd rather.

**Comment by rbehjati** [comment]
> Ah, it looks like you may have already changed those, so I can drop #45 if you'd rather.

I did not fix everything, for instance renaming `AuthLogicStatement`. So either way one of us has to rebase and fix merge conflicts.

**Comment by aferr** [comment]
> > Ah, it looks like you may have already changed those, so I can drop #45 if you'd rather.
> 
> I did not fix everything, for instance renaming `AuthLogicStatement`. So either way one of us has to rebase and fix merge conflicts.

I ran `golint` on the auth-logic` directory locally, so if you just `git checkout experimental/auth-logic/ -- 362698b31585bba35fbf8a4375f979e410c1990f` (where that hash is head of main), and then rebase onto #45, that subidrectory should lint.

File: experimental/auth-logic/provenance_build_wrapper_test.go [hunk 1]
```diff
@@ -61,4 +63,8 @@ func TestProvenanceBuildWrapper(t *testing.T) {
 		t.Errorf("got:\n%v\nwant:\n%v\n", got, want)
 	}
 
+	// cleanup
+	if err := os.RemoveAll("/tmp/release"); err != nil {
```
Comment (aferr) [inline experimental/auth-logic/provenance_build_wrapper_test.go:67]: Oh this is much better :)

**Review by aferr — APPROVED** [review]

File: experimental/auth-logic/provenance_build_wrapper_test.go (hunk 1 above)
Comment (rbehjati) [reply experimental/auth-logic/provenance_build_wrapper_test.go:67]: Does not seem like it. So I am using `os.TempDir` now. I wonder why it did not fail before on the CI.

**Review by tiziano88 — APPROVED** [review]

**Pull request merged**